tests/test_model_wave.py

```diff
@@ -1,25 +1,36 @@
 #! /usr/bin/env python
 import os
-import sys
-import time
+import unittest
 
+import astropy.units as u
 import numpy
-from astropy import log
 
-import pint.models as tm
-from pint import toa
+import pint.fitter
+import pint.models
+import pint.residuals
+import pint.toa
 from pinttestdata import datadir
 
+# Not included in the test here, but as a sanity check I used this same
+# ephemeris to phase up Fermi data, and it looks good.
 
-def test_wave():
-    os.chdir(datadir)
-    parfile = "J1513-5908_PKS_alldata_white.par"
-    timfile = "J1744-1134.Rcvr1_2.GASP.8y.x.tim"
-    t = toa.get_TOAs(timfile, usepickle=False)
+parfile = os.path.join(datadir, "vela_wave.par")
+timfile = os.path.join(datadir, "vela_wave.tim")
 
-    m = tm.get_model(parfile)
-    log.info("model.as_parfile():\n%s" % m.as_parfile())
 
-    print(m.as_parfile())
-    print(m)
-    print(m.delay(t))
+class TestWave(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.m = pint.models.get_model(parfile)
+        cls.t = pint.toa.get_TOAs(timfile, ephem="DE405", include_bipm=False)
+
+    def test_vela(self):
+        print("Test RMS of a VELA ephemeris with WAVE parameters.")
+        rs = pint.residuals.Residuals(self.t, self.m).time_resids
+        rms = rs.to(u.us).std()
+        emsg = "RMS of " + str(rms.value) + " is too big."
+        assert rms < 350.0 * u.us, emsg
```

Comment:
Is it worth including a version without WAVE and confirming that the RMS is worse?

Reply:
It would be great!

Reply:
With this example, without the waves there are many, many wraps, and the rms will be catastrophic.  So in that sense it's a fairly stringent test since the waves need to be working and working well to pass the test.  Of course, there is always room for more tests.

Based on the email exchange I've implemented Waves as a PhaseComponent and slightly tweaked the test.  I plan to submit this as a separate PR, so this one can be rejected/deleted, imo.

Reply:
When the new one is ready to upload, mention that it supersedes this one (by number so GitHub will make the link) and we can close this one.